### ISSUE TITLE
Can support default value?

### ISSUE THREAD

**Issue by kjsolo** [rharter/auto-value-gson#47]
I do not want use @Nullable

**Comment by rharter** [comment]
Can you please detail what you're asking for and why?

**Comment by kjsolo** [comment]
``java
class User {
    String name;
    String nickname;
}
```

Sometime service does not  pass nickname and application will crash, But I do not want use @Nullable and check "null" every time.

**Comment by Petrulak** [comment]
It would be pretty handy to have some default value  since i am facing same issue.
For example i am fetching list of object from API , and some object doesn't have certain property, so only way how to fix is it to annotate field with @Nullable or to alter **read** method in  **final class AutoValue_XYZ extends $AutoValue_XYZ** which is pretty nasty adn harder to maintain if i want to add new fields. @rharter are you planning to implement some kind of default value in future ?
Thanks

**Comment by autonomousapps** [comment]
The readme discusses support for default values, but I can't seem to get it to work. The following compiles:

```java
@AutoValue public abstract class Lesson {

    public abstract long id();
    public abstract String name();

    public static TypeAdapter<Lesson> typeAdapter(Gson gson) {
        return new AutoValue_Lesson.GsonTypeAdapter(gson);
    }
}
```

But this does not:

```java
@AutoValue public abstract class Lesson {

    public abstract long id();
    public abstract String name();

    public static TypeAdapter<Lesson> typeAdapter(Gson gson) {
        return new AutoValue_Lesson.GsonTypeAdapter(gson)
                .defaultId(0L)
                .defaultName("")
    }
}
```

The error looks like this:

```
Lesson.java:46: error: cannot find symbol
                .defaultName("");
```

I am including the library like so:

```groovy
    provided "com.ryanharter.auto.value:auto-value-gson:0.4.4"
    apt "com.ryanharter.auto.value:auto-value-gson:0.4.4"
```

**Comment by autonomousapps** [comment]
From the changelog, it looks like maybe default value support has not been released yet?

**Comment by rharter** [comment]
That's correct. Was waiting on a couple more to release.

**Comment by autonomousapps** [comment]
Do you have an estimate when that might be? Just curious. I can build locally if I have to.

**Comment by rharter** [comment]
I can take another look tonight

**Comment by autonomousapps** [comment]
Thank you!

**Comment by rharter** [comment]
I just released 0.4.5, it should be available soon.

**Comment by autonomousapps** [comment]
Thanks again! AutoValue, auto-value-parcel, and auto-value-gson are helping me cut the size of my value classes in half (and also enforce immutability).
